pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -385,9 +385,14 @@ func (r *RestartReconciler) doesDBHaveQuorum(clusterState map[string]string) boo
 // show up as down in the cluster state.  Even then, there is still a chance
 // that this may report a node is UP but not yet accepting connections because
 // it could doing the initialization phase.
-func (r *RestartReconciler) fetchClusterNodeStatus(ctx context.Context) (map[string]string, ctrl.Result, error) {
+func (r *RestartReconciler) fetchClusterNodeStatus(ctx context.Context, downPodsFact []*PodFact) (map[string]string, ctrl.Result, error) {
+	downPods := make([]string, 0, len(downPodsFact))
+	for i := range downPodsFact {
+		downPods = append(downPods, downPodsFact[i].podIP)
+	}
 	opts := []fetchnodestate.Option{
 		fetchnodestate.WithInitiator(r.InitiatorPod, r.InitiatorPodIP),
+		fetchnodestate.WithHosts(downPods),
```

Comment:
@spilchen The new fetchNodeState API use both nma and http endpoint instead of only http endpoint  https://git.verticacorp.com/projects/VER/repos/server/browse/platform/vcluster/vclusterops/fetch_node_state.go#177-186, It cause a lot of failures for e2e tests. I just provide certs for NMA to solve issue

Previously behavior, we only use http for only inital node and use http get all nodes state. Now, with the recent change, we need to configure all NMA certificates for all pods to use NMA read catalog editor endpoint to retrieve information. I just provide downPods to solve it, but let me know if we any existing function available to retrieve pods in the database?

Reply:
Shouldn't there be a change in `fetch_node_state_vc.go`? I understand that we need to pass in the certs (probably with a change to `fetch_node_state_vc.go`). But why do we have to pass in the down hosts?

Reply:
EDIT: now I see the `fetch_node_state_vc.go`. But I still don't understand why the down nodes are needed.

Reply:
I got timeout error for down nodes at NMAReadVerticaVersionOp  step in fech_node_state
```
controllers.VerticaDB.FetchNodeState.NMAReadVerticaVersionOp	[NMAReadVerticaVersionOp] result from host 10.244.0.215 summary FAILURE, details: fail to send request v1/vertica/version on host 10.244.0.215, details Get "https://10.244.0.215:5554/v1/vertica/version": dial tcp 10.244.0.215:5554: connect: connection timed out	{"verticadb": "kuttl-test-lasting-parrot/v-restart", "reconcile-uuid": "8075ac96-4a1d-4afb-941d-3602af402426"}
verticadb-operator verticadb-operator-manager-7d4bb66d4-9hfj8 manager github.com/vertica/vcluster/vclusterops/vlog.(*Printer).PrintError
```
It seems we don't have certs in down nodes

Reply:
I think this needs to do back to Le. Down nodes are probably running with the NMA, but the IPs in the catalog are different, so the endpoint request is timing out.

Reply:
In the meantime, do we need the latest change from vclusterops? I think the fetch node state change was only in the last commit. I wonder if we can pull in the vclusterOps changes from before the last commit. We still need to address this issue, but at least we can make some progress.